lib/validators/actions.js

```diff
@@ -120,6 +120,86 @@ export function runMetaValidators(uri, errorValidators, warningValidators) {
     .catch(log.error);
 }
 
+/**
+ * Check if the validationRule has the correct properties to run validation with
+ * @param {object} validators
+ * @param {object} validationRule
+ * @return {boolean}
+ */
+function validKilnjsValidator(validators, validationRule) {
+  return  validationRule.validator &&
+          validators[validationRule.validator] &&
+          validators[validationRule.validator].type &&
+          validators[validationRule.validator].kilnjsValidate;
+}
+
+/**
+ * Run the kilnjsValidators
+ * @param {object} schemas & components
+ * @param {string} errorLevel
+ * @return {array}
+ */
+export function runKilnjsValidators({schemas = {}, components}, errorLevel) {
+  const { validators } = window.kiln,
+    errors = [];
+
+  Object.keys(schemas).forEach((schemaName) => {
+    // Loop through all schemas on the page and check if they have a validation property
```

Comment:
In my understanding this can be implied from the code.

Reply:
Agree. I ended up rewriting this function and breaking it up into smaller ones, removing most of the comments as I did it.

Reply:
uffff looks much better now! Nice! 👍